internal/ingest/linksystem.go

```diff
@@ -204,35 +213,146 @@ func (ing *Ingester) ingestAd(publisherID peer.ID, adCid cid.Cid, ad schema.Adve
 
 	startTime := time.Now()
 
-	// Traverse entries based on the entries selector that limits recursion depth.
+	// The ad.Entries link can point to either a chain of EntryChunks or a HAMT.
+	// Sync the very first entry so that we can check which type it is.
+	// Note, this means the maximum depth of entries traversal will be 1 plus the configured max depth.
+	// TODO: See if it is worth detecting and reducing depth the depth in entries selectors by one.
```

Comment:
We _can_ detect if the selector has a depth configured and reduce the depth by one similar to the work I did [here](https://github.com/filecoin-project/go-legs/blob/main/selector.go#L78-L172). But I am not sure if it is worth doing. 

Left a TODO to revisit, later since regardless of whether it's worth doing or not I'd like to attend to it in a separate PR to reduce the scope of work here.

Reply:
todo for now seems fine